### PR TITLE
deps: update typescript to 5.3.3

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Keep the dependencies up-to-date. 

**Description**

Update Typescript to latest 5.3.3.

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/prover/src/utils/consensus.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export async function fetchNearestBlock(
 ): Promise<capella.SignedBeaconBlock> {
   const res = await api.beacon.getBlockV2(slot);
 
-  if (res.ok) return res.response.data;
+  if (res.ok) return res.response.data as capella.SignedBeaconBlock;
```
Comment (jeluard) [inline packages/prover/src/utils/consensus.ts:16]: The new typescript found this type issue?

**Review by jeluard — DISMISSED** [review]

File: packages/prover/src/utils/consensus.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/prover/src/utils/consensus.ts:16]: yes the newer version is strict when returning union type.

**Comment by nazarhussain** [comment]
For reference the biggest change in this version is actually some technical debt on the Typescript side. 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html#consolidation-between-tsserverlibraryjs-and-typescriptjs

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.17.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.17.0) :tada:

**Pull request merged**